### ISSUE TITLE
Cannot copy if one RTDC file is corrupt

### ISSUE THREAD

**Issue by B-Hartmann** [GuckLab/MPL-Data-Cast#13]
MPL data cast version 0.2.3
Windows 10
Python 3.9.9

I know that one of my .rtdc files is kind of broken, because Shape-In leaves the file in a broken state when you start a fluorescence measurement and the lids of the device are open. The interlock causes the measurement to be aborted and Shape-In creates a file that contains no event data (see Shape-In [issue 15](https://github.com/ZELLMECHANIK-DRESDEN/ShapeIn_Issues/issues/15)).

When trying to copy the data, I get the following error:
```
(mpl_data_cast) C:\Users\labuser>mpldc cast -r RTDC "D:\Experiments\20220407_BHCS_HeLa_Enucleation" "T:\Members\Benedikt\mpl_data_cast_test_ordner\20220407_BHCS_HeLa_Enucleation"
Using recipe RTDC.
Processed 1 files (~0.2 MB/s).
Traceback (most recent call last):
  File "C:\Users\labuser\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\labuser\AppData\Local\Programs\Python\Python39\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\Users\labuser\AppData\Local\Programs\Python\Python39\Scripts\mpldc.exe\__main__.py", line 7, in <module>
  File "C:\Users\labuser\AppData\Local\Programs\Python\Python39\lib\site-packages\click\core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "C:\Users\labuser\AppData\Local\Programs\Python\Python39\lib\site-packages\click\core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "C:\Users\labuser\AppData\Local\Programs\Python\Python39\lib\site-packages\click\core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "C:\Users\labuser\AppData\Local\Programs\Python\Python39\lib\site-packages\click\core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "C:\Users\labuser\AppData\Local\Programs\Python\Python39\lib\site-packages\click\core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "C:\Users\labuser\AppData\Local\Programs\Python\Python39\lib\site-packages\mpl_data_cast\cli.py", line 79, in cast
    rp.cast(path_callback=path_callback, **kwargs)
  File "C:\Users\labuser\AppData\Local\Programs\Python\Python39\lib\site-packages\mpl_data_cast\recipe.py", line 65, in cast
    self.convert_dataset(path_list=path_list, temp_path=temp_path,
  File "C:\Users\labuser\AppData\Local\Programs\Python\Python39\lib\site-packages\mpl_data_cast\mod_recipes\rcp_rtdc.py", line 13, in convert_dataset
    dclab.cli.compress(path_out=temp_path, path_in=path_list[0])
  File "C:\Users\labuser\AppData\Local\Programs\Python\Python39\lib\site-packages\dclab\cli\task_compress.py", line 45, in compress
    ds.export.hdf5(path=path_temp,
  File "C:\Users\labuser\AppData\Local\Programs\Python\Python39\lib\site-packages\dclab\rtdc_dataset\export.py", line 243, in hdf5
    store_filtered_feature(rtdc_writer=hw,
  File "C:\Users\labuser\AppData\Local\Programs\Python\Python39\lib\site-packages\dclab\rtdc_dataset\writer.py", line 63, in __exit__
    self.rectify_metadata()
  File "C:\Users\labuser\AppData\Local\Programs\Python\Python39\lib\site-packages\dclab\rtdc_dataset\writer.py", line 113, in rectify_metadata
    shape = self.h5file["events"]["image"][0].shape
  File "h5py\_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py\_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "C:\Users\labuser\AppData\Local\Programs\Python\Python39\lib\site-packages\h5py\_hl\dataset.py", line 710, in __getitem__
    return self._fast_reader.read(args)
  File "h5py\_selector.pyx", line 351, in h5py._selector.Reader.read
  File "h5py\_selector.pyx", line 151, in h5py._selector.Selector.apply_args
IndexError: Index (0) out of range for empty dimension

```

**Comment by B-Hartmann** [comment]
Since the broken file was the very first in the first folder that was copied, the error happened very quickly, but imagine you start copying 100 GB of data and after 2/3 of the process, this happens. Once there is a GUI, I think a job management like in DCOR-Aid would be nice, so unfinished jobs are taken care of when there is an error.

**Comment by paulmueller** [comment]
Thanks for testing! There should be some mechanism in the library and in the CLI that ignores errors and gives the user visual feedback when an exception occurred.

**Comment by paulmueller** [comment]
fixed in 0.2.4
